backends/dpdk/dpdkArch.cpp

```diff
@@ -0,0 +1,920 @@
+      auto parser = kv.first->to<IR::P4Parser>();
+      auto local_metadata = parser->getApplyParameters()->getParameter(2);
+      local_metadata_type =
+          local_metadata->type->to<IR::Type_Name>()->path->name;
+      auto header = parser->getApplyParameters()->getParameter(1);
+      header_type = header->type->to<IR::Type_Name>()->path->name;
+      pushMetadata(parser->getApplyParameters()->getParameter(2));
+      pushMetadata(parser->getApplyParameters()->getParameter(3));
+      pushMetadata(parser->getApplyParameters()->getParameter(4));
+      pushMetadata(parser->getApplyParameters()->getParameter(5));
+    } else if (kv.second.pipe == "Ingress") {
+      auto control = kv.first->to<IR::P4Control>();
+      pushMetadata(control->getApplyParameters()->getParameter(1));
+      pushMetadata(control->getApplyParameters()->getParameter(2));
+      pushMetadata(control->getApplyParameters()->getParameter(3));
+    } else if (kv.second.pipe == "IngressParser") {
+      auto deparser = kv.first->to<IR::P4Control>();
+      pushMetadata(deparser->getApplyParameters()->getParameter(1));
+      pushMetadata(deparser->getApplyParameters()->getParameter(2));
+      pushMetadata(deparser->getApplyParameters()->getParameter(3));
+      pushMetadata(deparser->getApplyParameters()->getParameter(5));
+      pushMetadata(deparser->getApplyParameters()->getParameter(6));
+    } else if (kv.second.pipe == "EgressParser") {
+      auto parser = kv.first->to<IR::P4Parser>();
+      pushMetadata(parser->getApplyParameters()->getParameter(2));
+      pushMetadata(parser->getApplyParameters()->getParameter(3));
+      pushMetadata(parser->getApplyParameters()->getParameter(4));
+      pushMetadata(parser->getApplyParameters()->getParameter(5));
+      pushMetadata(parser->getApplyParameters()->getParameter(6));
+    } else if (kv.second.pipe == "Egress") {
+      auto control = kv.first->to<IR::P4Control>();
+      pushMetadata(control->getApplyParameters()->getParameter(1));
+      pushMetadata(control->getApplyParameters()->getParameter(2));
+      pushMetadata(control->getApplyParameters()->getParameter(3));
+    } else if (kv.second.pipe == "EgressDeparser") {
+      auto deparser = kv.first->to<IR::P4Control>();
+      pushMetadata(deparser->getApplyParameters()->getParameter(1));
+      pushMetadata(deparser->getApplyParameters()->getParameter(2));
+      pushMetadata(deparser->getApplyParameters()->getParameter(4));
+      pushMetadata(deparser->getApplyParameters()->getParameter(5));
+      pushMetadata(deparser->getApplyParameters()->getParameter(6));
+    }
+  }
+  return true;
+}
+
+bool CollectMetadataHeaderInfo::preorder(const IR::Type_Struct *s) {
+  for (auto m : used_metadata) {
+    if (m->to<IR::Type_Name>()->path->name.name == s->name.name) {
+      for (auto field : s->fields) {
+        fields.push_back(new IR::StructField(
+            IR::ID(TypeStruct2Name(s->name.name) + "_" + field->name),
```

Comment:
@gordonwucn please take a look.

Reply:
This place will not cause conflicts. For intrinsic metadata, it prepends the metadata header name to variable name(in the form m.intrinsic_header_name_variable_name). For user define metadata, it prepends the "local_metadata_" to variable name(in the form m.local_metadata_variable_name). In this way, it will not cause conflicts.